node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -285,6 +289,107 @@ var deletePrincipal = module.exports.deletePrincipal = function(principalId, cal
+        if (err) {
+            return callback(err);
+        }
+
+        // Get the login to delete the user form the table AuthenticationLoginId
+        require('oae-authentication').getUserLoginIds(ctx, user.id, function(err, login){
+            if (err) {
+                return callback(err);
+            }
+
+            // Set (or re-Set) the principal as deleted in the authz index
+            AuthzDelete.setDeleted(user.id, function(err) {
+                if (err) {
+                    return callback(err);
+                }
+
+                // Delete user from table Principals
+                Cassandra.runQuery('DELETE FROM "Principals" where "principalId" = ?', [user.id], function(err, rows) {
+                    if (err) {
+                        return callback(err);
+                    }
+
+                    // Delete user from table PrincipalsByEmail
+                    Cassandra.runQuery('DELETE FROM "PrincipalsByEmail" where "email" = ? AND "principalId" = ?', [user.email, user.id], function(err, rows) {
+                        if (err) {
+                            return callback(err);
+                        }
+                    
+                        var loginId = new LoginId(user.tenant.alias, AuthenticationConstants.providers.LOCAL, login.local);
+
+                        // Delete user from table AuthenticationUserLoginId
+                        Cassandra.runQuery('DELETE FROM "AuthenticationUserLoginId" WHERE "userId" = ?', [user.id], function(err, rows) {
+                            if (err) {
+                                return callback(err);
+                            }
+
+                            // Delete user from table AuthenticationLoginId
+                            if(loginId.externalId !== undefined){
+                                Cassandra.runQuery('DELETE FROM "AuthenticationLoginId" WHERE "loginId" = ?', [_flattenLoginId(loginId)], function(err, rows) {
+                                    if (err) {
+                                        return callback(err);
+                                    }
+
+                                    // Update the cache,
+                                    return OaeUtil.invokeIfNecessary(isUser(user.id), _updateCachedUser, user.id, {'deleted': deleted}, callback);
+                                });
+                            }
+                        });
+                    });
+                });
+            });
+        });
+    });
+};
+
+/**
+ * Delete a group
+ *
+ * @param  {String}     group           The group
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var deleteGroupFromPrincipals = module.exports.deleteGroupFromPrincipals = function(group, callback){
+    // Delete user from table Principals
+    Cassandra.runQuery('DELETE FROM "Principals" where "principalId" = ?', [group.id], function(err, rows) {
```

Comment:
I would also check that there are no other tables that would now have a reference to a deleted group.